File: igor/igor-core/src/main/java/com/netflix/spinnaker/igor/keel/KeelService.java
/*
 * Copyright 2019 Netflix, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.netflix.spinnaker.igor.keel;

import java.util.Map;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface KeelService {
  /**
   * Events should be sent with this format (inherited from echo events): [ payload: [artifacts:
   * List<Artifact>, details: Map], eventName: String ]
   */
  @POST("artifacts/events")
  Call<Void> sendArtifactEvent(@Body Map event);
}

File: orca/orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/KeelService.kt
/*
 * Copyright 2017 Netflix, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License")
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.netflix.spinnaker.orca

import com.netflix.spinnaker.orca.keel.model.DeliveryConfig
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.Headers
import retrofit2.http.POST
import retrofit2.http.Path


interface KeelService {
  @POST("/delivery-configs/")
  @Headers("Accept: application/json")
  fun publishDeliveryConfig(@Body deliveryConfig: DeliveryConfig): Call<ResponseBody>

  @DELETE("/application/{application}/config")
  fun deleteDeliveryConfig(@Path("application") application: String): Call<ResponseBody>
}
